frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/vm-settings-tab-state-update.ts

```diff
@@ -156,11 +156,22 @@ const osUpdater = ({ id, prevState, dispatch, getState }: UpdateOptions) => {
   const isWindows = os?.startsWith('win');
 
   dispatch(
-    vmWizardInternalActions[InternalActionType.UpdateVmSettingsField](
-      id,
-      VMSettingsField.MOUNT_WINDOWS_GUEST_TOOLS,
-      { isHidden: asHidden(!isWindows, VMSettingsField.OPERATING_SYSTEM), value: isWindows },
-    ),
+    vmWizardInternalActions[InternalActionType.UpdateVmSettings](id, {
+      [VMSettingsField.MOUNT_WINDOWS_GUEST_TOOLS]: {
+        isHidden: asHidden(!isWindows, VMSettingsField.OPERATING_SYSTEM),
+        value: isWindows,
+      },
+      [VMSettingsField.WORKLOAD_PROFILE]: { value: null },
```

Comment:
additional idea: we could also show a small warning to the user that a combination is invalid and reset the warning on any new value selection

thoughts?

Reply:
I agree with the first comment.
for the warning I think it's redundant. when following the first comment behavior, it is clear that if the workload/flavor would set to null it is because the combination is invalid, thus no need for any warning message.

what do you think @matthewcarleton  ?

Reply:
Note:
two os may have the same flavor label, but it may mean different things, for example:
flavor: tiny os:ubuntu => mem: 2Gi
flavor: tiny os:fedora => mem: 1Gi

IMHO, not re-seting the flavor each time we change an os is not nice to users 

https://github.com/kubevirt/common-templates/blob/e127881c7fb7f77ba84f0571478d4507e3e6b8e7/generate-templates.yaml#L204
https://github.com/kubevirt/common-templates/blob/e127881c7fb7f77ba84f0571478d4507e3e6b8e7/generate-templates.yaml#L162

Reply:
why is it not nice? the user just sees the label tiny and probably does not know which template will be used, ie does not have an association with how much memory. And if he does know, it shouldn't matter either. The user is selecting a tiny flavor for a reason.

Reply:
> why is it not nice? 

All the times I ordered a VM, I used numbers (or looked up doc to see the numbers) to specify the mem/cpu/disk I needed , I think that a user like me, will be confused if they choose 2Gb (tiny on ubuntu) and got 1Gb (tiny on fedora) while they should have chosen small because it's the 2Gb for fedora.

But I don't feel strongly about this, so just ignore if you feel differently, the bug is about having the ability to choose any OS, so any fix that will show all the OS options for any given flavour/workload is good for me :-)

Reply:
I know it is a bit confusing, I just meant it in the context of this PR. Sure if we plan to redesign this feature, we should tackle this problem in a separate effort. As this concerns the basic flow as well, and not just this fix.